<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=11"/>
<meta name="generator" content="Doxygen 1.9.6"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>RNAlib-2.6.0b: Bibliography</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
  $(document).ready(function() { init_search(); });
/* @license-end */
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr id="projectrow">
  <td id="projectalign">
   <div id="projectname">RNAlib-2.6.0b
   </div>
  </td>
    <td>        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <span id="MSearchSelect"                onmouseover="return searchBox.OnSearchSelectShow()"                onmouseout="return searchBox.OnSearchSelectHide()">&#160;</span>
          <input type="text" id="MSearchField" value="" placeholder="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.svg" alt=""/></a>
          </span>
        </div>
</td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.9.6 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
var searchBox = new SearchBox("searchBox", "search/",'.html');
/* @license-end */
</script>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
$(document).ready(function(){initNavTree('citelist.html',''); initResizable(); });
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<div id="MSearchResults">
<div class="SRPage">
<div id="SRIndex">
<div id="SRResults"></div>
<div class="SRStatus" id="Loading">Loading...</div>
<div class="SRStatus" id="Searching">Searching...</div>
<div class="SRStatus" id="NoMatches">No Matches</div>
</div>
</div>
</div>
</div>

<div><div class="header">
  <div class="headertitle"><div class="title">Bibliography </div></div>
</div><!--header-->
<div class="contents">
<div class="textblock"><dl class="citelist">
<dt><a class="anchor" id="CITEREF_bernhart:2005"></a>[1]</dt>
<dd><p class="startdd">Stephan&#160;H Bernhart, Ivo&#160;L Hofacker, and Peter&#160;F Stadler. Local RNA base pairing probabilities in large sequences. <em>Bioinformatics</em>, 22(5):614&ndash;615, 2005.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_bernhart:2006"></a>[2]</dt>
<dd><p class="startdd">S.H. Bernhart, H.&#160;Tafer, U.&#160;M&uuml;ckstein, C.&#160;Flamm, P.F. Stadler, and I.L. Hofacker. Partition function and base pairing probabilities of RNA heterodimers. <em>Algorithms for Molecular Biology</em>, 1(1):3, 2006.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_bernhart:2008"></a>[3]</dt>
<dd><p class="startdd">S.H. Bernhart, I.L. Hofacker, S.&#160;Will, A.R. Gruber, and P.F. Stadler. RNAalifold: Improved consensus structure prediction for RNA alignments. <em>BMC bioinformatics</em>, 9(1):474, 2008.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_bernhart:2011"></a>[4]</dt>
<dd><p class="startdd">Stephan&#160;H Bernhart, Ullrike M&uuml;ckstein, and Ivo&#160;L Hofacker. RNA accessibility in cubic time. <em>Algorithms for Molecular Biology</em>, 6(1):3, 2011.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_boccaletto:2022"></a>[5]</dt>
<dd><p class="startdd">Pietro Boccaletto, Filip Stefaniak, Angana Ray, Andrea Cappannini, Sunandan Mukherjee, Elzbieta Purta, Malgorzata Kurkowska, Niloofar Shirvanizadeh, Eliana Destefanis, Paula Groza, and others. MODOMICS: a database of RNA modification pathways. 2021 update. <em>Nucleic Acids Research</em>, 50(D1):D231&ndash;D235, 2022. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_bruccoleri:1988"></a>[6]</dt>
<dd><p class="startdd">R.E. Bruccoleri and G.&#160;Heinrich. An improved algorithm for nucleic acid secondary structure display. <em>Computer applications in the biosciences: CABIOS</em>, 4(1):167&ndash;173, 1988.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_dalluge:1996"></a>[7]</dt>
<dd><p class="startdd">Joseph&#160;J Dalluge, Takeshi Hashizume, Alan&#160;E Sopchik, James&#160;A McCloskey, and Darrell&#160;R Davis. Conformational flexibility in RNA: the role of dihydrouridine. <em>Nucleic acids research</em>, 24(6):1073&ndash;1079, 1996. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_deigan:2009"></a>[8]</dt>
<dd><p class="startdd">Katherine&#160;E. Deigan, Tian&#160;W. Li, David&#160;H. Mathews, and Kevin&#160;M. Weeks. Accurate SHAPE-directed RNA structure determination. <em>PNAS</em>, 106:97&ndash;102, 2009.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_flamm:2001"></a>[9]</dt>
<dd><p class="startdd">Christoph Flamm, Ivo&#160;L Hofacker, Sebastian Maurer-Stroh, Peter&#160;F Stadler, and Martin Zehl. Design of multistable RNA molecules. <em>RNA</em>, 7(02):254&ndash;265, 2001.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_fontana:1993b"></a>[10]</dt>
<dd><p class="startdd">W.&#160;Fontana, P.F. Stadler, E.G. Bornberg-Bauer, T.&#160;Griesmacher, I.L. Hofacker, M.&#160;Tacker, P.&#160;Tarazona, E.D. Weinberger, and P.&#160;Schuster. RNA folding and combinatory landscapes. <em>Physical review E</em>, 47(3):2083, 1993.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_freyhult:2005"></a>[11]</dt>
<dd><p class="startdd">Eva Freyhult, Vincent Moulton, and Paul Gardner. Predicting RNA structure using mutual information. <em>Applied bioinformatics</em>, 4(1):53&ndash;59, 2005.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_giegerich:2004"></a>[12]</dt>
<dd><p class="startdd">Robert Giegerich, Bj&ouml;rn Vo&szlig;, and Marc Rehmsmeier. Abstract shapes of RNA. <em>Nucleic Acids Research</em>, 32(16):4843&ndash;4851, 2004.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_hofacker:2006"></a>[13]</dt>
<dd><p class="startdd">I.L. Hofacker and P.F. Stadler. Memory efficient folding algorithms for circular RNA secondary structures. <em>Bioinformatics</em>, 22(10):1172&ndash;1176, 2006.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_hofacker:1994"></a>[14]</dt>
<dd><p class="startdd">I.L. Hofacker, W.&#160;Fontana, P.F. Stadler, L.S. Bonhoeffer, M.&#160;Tacker, and P.&#160;Schuster. Fast folding and comparison of RNA secondary structures. <em>Monatshefte f&uuml;r Chemie/Chemical Monthly</em>, 125(2):167&ndash;188, 1994.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_hofacker:2002"></a>[15]</dt>
<dd><p class="startdd">I.L. Hofacker, M.&#160;Fekete, and P.F. Stadler. Secondary structure prediction for aligned RNA sequences. <em>Journal of molecular biology</em>, 319(5):1059&ndash;1066, 2002.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_hudson:2013"></a>[16]</dt>
<dd><p class="startdd">Graham&#160;A Hudson, Richard&#160;J Bloomingdale, and Brent&#160;M Znosko. Thermodynamic contribution and nearest-neighbor parameters of pseudouridine-adenosine base pairs in oligoribonucleotides. <em>RNA</em>, 19(11):1474&ndash;1482, 2013. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_jolley:2017"></a>[17]</dt>
<dd><p class="startdd">Elizabeth&#160;A Jolley and Brent&#160;M Znosko. The loss of a hydrogen bond: Thermodynamic contributions of a non-standard nucleotide. <em>Nucleic acids research</em>, 45(3):1479&ndash;1487, 2017. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_kierzek:2022"></a>[18]</dt>
<dd><p class="startdd">Elzbieta Kierzek, Xiaoju Zhang, Richard&#160;M Watson, Scott&#160;D Kennedy, Marta Szabat, Ryszard Kierzek, and David&#160;H Mathews. Secondary structure prediction for RNA sequences including N6-methyladenosine. <em>Nature communications</em>, 13(1):1&ndash;10, 2022. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_lorenz:2009"></a>[19]</dt>
<dd><p class="startdd">Ronny Lorenz, Christoph Flamm, and Ivo&#160;L. Hofacker. 2d projections of RNA folding landscapes. In Ivo Grosse, Steffen Neumann, Stefan Posch, Falk Schreiber, and Peter&#160;F. Stadler, editors, <em>German Conference on Bioinformatics 2009</em>, volume 157 of <em>Lecture Notes in Informatics</em>, pages 11&ndash;20, Bonn, September 2009. Gesellschaft f. Informatik.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_lorenz:2011"></a>[20]</dt>
<dd><p class="startdd">Ronny Lorenz, Stephan&#160;H. Bernhart, Christian H&ouml;ner&#160;zu Siederdissen, Hakim Tafer, Christoph Flamm, Peter&#160;F. Stadler, and Ivo&#160;L. Hofacker. ViennaRNA package 2.0. <em>Algorithms for Molecular Biology</em>, 6(1):26, 2011. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_lorenz:2016b"></a>[21]</dt>
<dd><p class="startdd">Ronny Lorenz, Ivo&#160;L. Hofacker, and Peter&#160;F. Stadler. RNA folding with hard and soft constraints. <em>Algorithms for Molecular Biology</em>, 11(1):1&ndash;13, 2016. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_lorenz:2016a"></a>[22]</dt>
<dd><p class="startdd">Ronny Lorenz, Dominik Luntzer, Ivo&#160;L. Hofacker, Peter&#160;F. Stadler, and Michael&#160;T. Wolfinger. <a href="http://bioinformatics.oxfordjournals.org/content/32/1/145.abstract">Shape directed rna folding</a>. <em>Bioinformatics</em>, 32(1):145&ndash;147, 2016. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_mccaskill:1990"></a>[23]</dt>
<dd><p class="startdd">J.S. McCaskill. The equilibrium partition function and base pair binding probabilities for RNA secondary structure. <em>Biopolymers</em>, 29(6-7):1105&ndash;1119, 1990.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_michalik:2017"></a>[24]</dt>
<dd><p class="startdd">Juraj Mich&aacute;lik, H&eacute;l&egrave;ne Touzet, and Yann Ponty. Efficient approximations of RNA kinetics landscape using non-redundant sampling. <em>Bioinformatics</em>, 33(14):i283&ndash;i292, 2017.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_richardson:2016"></a>[25]</dt>
<dd><p class="startdd">Katherine&#160;E Richardson and Brent&#160;M Znosko. Nearest-neighbor parameters for 7-deaza-adenosine textperiodcentered uridine base pairs in RNA duplexes. <em>RNA</em>, 22(6):934&ndash;942, 2016. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_sawada:2003"></a>[26]</dt>
<dd><p class="startdd">Joe Sawada. A fast algorithm to generate necklaces with fixed content. <em>Theoretical Computer Science</em>, 301(1):477&ndash;489, 2003.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_shapiro:1990"></a>[27]</dt>
<dd><p class="startdd">B.A. Shapiro and K.&#160;Zhang. Comparing multiple RNA secondary structures using tree comparisons. <em>Computer applications in the biosciences: CABIOS</em>, 6(4):309&ndash;318, 1990.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_shapiro:1988"></a>[28]</dt>
<dd><p class="startdd">B.A. Shapiro. An algorithm for comparing multiple RNA secondary structures. <em>Computer applications in the biosciences: CABIOS</em>, 4(3):387&ndash;393, 1988.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_washietl:2012"></a>[29]</dt>
<dd><p class="startdd">Stefan Washietl, Ivo&#160;L. Hofacker, Peter&#160;F. Stadler, and Manolis Kellis. RNA folding with soft constraints: reconciliation of probing data and thermodynamics secondary structure prediction. <em>Nucleic Acids Research</em>, 40(10):4261&ndash;4272, 2012.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_wiegreffe:2018"></a>[30]</dt>
<dd><p class="startdd">Daniel Wiegreffe, Daniel Alexander, Peter&#160;F Stadler, and Dirk Zeckzer. RNApuzzler: efficient outerplanar drawing of RNA-secondary structures. <em>Bioinformatics</em>, 2018.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_wright:2007"></a>[31]</dt>
<dd><p class="startdd">Daniel&#160;J Wright, Jamie&#160;L Rice, Dawn&#160;M Yanker, and Brent&#160;M Znosko. Nearest Neighbor Parameters for Inosine textperiodcentered Uridine Pairs in RNA Duplexes. <em>Biochemistry</em>, 46(15):4625&ndash;4634, 2007. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_wright:2018"></a>[32]</dt>
<dd><p class="startdd">Daniel&#160;J Wright, Christopher&#160;R Force, and Brent&#160;M Znosko. Stability of RNA duplexes containing inosine textperiodcentered cytosine pairs. <em>Nucleic Acids Research</em>, 46(22):12099&ndash;12108, 2018. </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_wuchty:1999"></a>[33]</dt>
<dd><p class="startdd">S.&#160;Wuchty, W.&#160;Fontana, I.&#160;L. Hofacker, and P.&#160;Schuster. Complete suboptimal folding of RNA and the stability of secondary structures. <em>Biopolymers</em>, 49(2):145&ndash;165, February 1999.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_zarringhalam:2012"></a>[34]</dt>
<dd><p class="startdd">Kourosh Zarringhalam, Michelle&#160;M. Meyer, Ivan Dotu, Jeffrey&#160;H. Chuang, and Peter Clote. Integrating chemical footprinting data into RNA secondary structure prediction. <em>PLOS ONE</em>, 7(10), 2012.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_zuker:1981"></a>[35]</dt>
<dd><p class="startdd">M.&#160;Zuker and P.&#160;Stiegler. Optimal computer folding of large RNA sequences using thermodynamics and auxiliary information. <em>Nucleic acids research</em>, 9(1):133&ndash;148, 1981.</p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="CITEREF_zuker:1989"></a>[36]</dt>
<dd><p class="startdd">M.&#160;Zuker. On finding all suboptimal foldings of an RNA molecule. <em>Science</em>, 244(4900):48&ndash;52, April 1989.</p>
<p class="enddd"></p>
</dd>
</dl>
</div></div><!-- contents -->
</div><!-- PageDoc -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Sun Apr 2 2023 13:14:55 for RNAlib-2.6.0b by <a href="https://www.doxygen.org/index.html"><img class="footer" src="doxygen.svg" width="104" height="31" alt="doxygen"/></a> 1.9.6 </li>
  </ul>
</div>
</body>
</html>
