<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=11"/>
<meta name="generator" content="Doxygen 1.9.6"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>RNAlib-2.6.0b: File Formats</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
  $(document).ready(function() { init_search(); });
/* @license-end */
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr id="projectrow">
  <td id="projectalign">
   <div id="projectname">RNAlib-2.6.0b
   </div>
  </td>
    <td>        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <span id="MSearchSelect"                onmouseover="return searchBox.OnSearchSelectShow()"                onmouseout="return searchBox.OnSearchSelectHide()">&#160;</span>
          <input type="text" id="MSearchField" value="" placeholder="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.svg" alt=""/></a>
          </span>
        </div>
</td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.9.6 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
var searchBox = new SearchBox("searchBox", "search/",'.html');
/* @license-end */
</script>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
$(document).ready(function(){initNavTree('file_formats.html',''); initResizable(); });
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<div id="MSearchResults">
<div class="SRPage">
<div id="SRIndex">
<div id="SRResults"></div>
<div class="SRStatus" id="Loading">Loading...</div>
<div class="SRStatus" id="Searching">Searching...</div>
<div class="SRStatus" id="NoMatches">No Matches</div>
</div>
</div>
</div>
</div>

<div><div class="header">
  <div class="headertitle"><div class="title">File Formats </div></div>
</div><!--header-->
<div class="contents">
<div class="toc"><h3>Table of Contents</h3>
<ul><li class="level1"><a href="#msa-formats">File formats for Multiple Sequence Alignments (MSA)</a><ul><li class="level2"><a href="#msa-formats-clustal">ClustalW format</a></li>
<li class="level2"><a href="#msa-formats-stockholm">Stockholm 1.0 format</a></li>
<li class="level2"><a href="#msa-formats-fasta">FASTA (Pearson) format</a></li>
<li class="level2"><a href="#msa-formats-maf">MAF format</a></li>
</ul>
</li>
<li class="level1"><a href="#constraint-formats">File formats to manipulate the RNA folding grammar</a><ul><li class="level2"><a href="#constraint-formats-file">Command Files</a><ul><li class="level3"><a href="#constraint_commands">Constraint commands</a></li>
<li class="level3"><a href="#domain_commands">RNA folding grammar exensions</a></li>
<li class="level3"><a href="#command_file_loop_types">Specification of the loop type context</a></li>
<li class="level3"><a href="#const_file_orientation">Controlling the orientation of base pairing</a></li>
<li class="level3"><a href="#const_file_seq_coords">Sequence coordinates</a></li>
<li class="level3"><a href="#const_file_syntax">Valid constraint commands</a></li>
<li class="level3"><a href="#domains_syntax">Valid domain extensions commands</a></li>
</ul>
</li>
</ul>
</li>
<li class="level1"><a href="#energy-parameter-formats">File Formats for Energy Parameters</a><ul><li class="level2"><a href="#modified-bases-params">JSON Parameter Files for Modified Bases</a></li>
</ul>
</li>
</ul>
</div>
<div class="textblock"><h1><a class="anchor" id="msa-formats"></a>
File formats for Multiple Sequence Alignments (MSA)</h1>
<h2><a class="anchor" id="msa-formats-clustal"></a>
ClustalW format</h2>
<p>The <em>ClustalW</em> format is a relatively simple text file containing a single multiple sequence alignment of DNA, RNA, or protein sequences. It was first used as an output format for the <em>clustalw</em> programs, but nowadays it may also be generated by various other sequence alignment tools. The specification is straight forward:</p>
<ul>
<li>The first line starts with the words <pre class="fragment">CLUSTAL W </pre> or <pre class="fragment">CLUSTALW </pre></li>
<li>After the above header there is at least one empty line</li>
<li>Finally, one or more blocks of sequence data are following, where each block is separated by at least one empty line</li>
</ul>
<p>Each line in a blocks of sequence data consists of the sequence name followed by the sequence symbols, separated by at least one whitespace character. Usually, the length of a sequence in one block does not exceed 60 symbols. Optionally, an additional whitespace separated cumulative residue count may follow the sequence symbols. Optionally, a block may be followed by a line depicting the degree of conservation of the respective alignment columns.</p>
<dl class="section note"><dt>Note</dt><dd>Sequence names and the sequences must not contain whitespace characters! Allowed gap symbols are the hyphen <em></em>("-"), and dot <em></em>(".").</dd></dl>
<dl class="section warning"><dt>Warning</dt><dd>Please note that many programs that output this format tend to truncate the sequence names to a limited number of characters, for instance the first 15 characters. This can destroy the uniqueness of identifiers in your MSA.</dd></dl>
<p>Here is an example alignment in ClustalW format: </p><pre class="fragment">CLUSTAL W (1.83) multiple sequence alignment


AL031296.1/85969-86120      CUGCCUCACAACGUUUGUGCCUCAGUUACCCGUAGAUGUAGUGAGGGUAACAAUACUUAC
AANU01225121.1/438-603      CUGCCUCACAACAUUUGUGCCUCAGUUACUCAUAGAUGUAGUGAGGGUGACAAUACUUAC
AAWR02037329.1/29294-29150  ---CUCGACACCACU---GCCUCGGUUACCCAUCGGUGCAGUGCGGGUAGUAGUACCAAU

AL031296.1/85969-86120      UCUCGUUGGUGAUAAGGAACAGCU
AANU01225121.1/438-603      UCUCGUUGGUGAUAAGGAACAGCU
AAWR02037329.1/29294-29150  GCUAAUUAGUUGUGAGGACCAACU 
</pre><h2><a class="anchor" id="msa-formats-stockholm"></a>
Stockholm 1.0 format</h2>
<p>Here is an example alignment in Stockholm 1.0 format: </p><pre class="fragment"># STOCKHOLM 1.0

#=GF AC   RF01293
#=GF ID   ACA59
#=GF DE   Small nucleolar RNA ACA59
#=GF AU   Wilkinson A
#=GF SE   Predicted; WAR; Wilkinson A
#=GF SS   Predicted; WAR; Wilkinson A
#=GF GA   43.00
#=GF TC   44.90
#=GF NC   40.30
#=GF TP   Gene; snRNA; snoRNA; HACA-box;
#=GF BM   cmbuild -F CM SEED
#=GF CB   cmcalibrate --mpi CM
#=GF SM   cmsearch --cpu 4 --verbose --nohmmonly -E 1000 -Z 549862.597050 CM SEQDB
#=GF DR   snoRNABase; ACA59;
#=GF DR   SO; 0001263; ncRNA_gene;
#=GF DR   GO; 0006396; RNA processing;
#=GF DR   GO; 0005730; nucleolus;
#=GF RN   [1]
#=GF RM   15199136
#=GF RT   Human box H/ACA pseudouridylation guide RNA machinery.
#=GF RA   Kiss AM, Jady BE, Bertrand E, Kiss T
#=GF RL   Mol Cell Biol. 2004;24:5797-5807.
#=GF WK   Small_nucleolar_RNA
#=GF SQ   3


AL031296.1/85969-86120     CUGCCUCACAACGUUUGUGCCUCAGUUACCCGUAGAUGUAGUGAGGGUAACAAUACUUACUCUCGUUGGUGAUAAGGAACAGCU
AANU01225121.1/438-603     CUGCCUCACAACAUUUGUGCCUCAGUUACUCAUAGAUGUAGUGAGGGUGACAAUACUUACUCUCGUUGGUGAUAAGGAACAGCU
AAWR02037329.1/29294-29150 ---CUCGACACCACU---GCCUCGGUUACCCAUCGGUGCAGUGCGGGUAGUAGUACCAAUGCUAAUUAGUUGUGAGGACCAACU
#=GC SS_cons               -----((((,&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;___________&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;,,,,&lt;&lt;&lt;&lt;&lt;&lt;&lt;______&gt;&gt;&gt;&gt;&gt;&gt;&gt;,,,,,))))::::::::::::
#=GC RF                    CUGCcccaCAaCacuuguGCCUCaGUUACcCauagguGuAGUGaGgGuggcAaUACccaCcCucgUUgGuggUaAGGAaCAgCU
//
</pre><dl class="section see"><dt>See also</dt><dd><a class="el" href="rna_structure_notations.html#wuss-notation">Washington University Secondary Structure (WUSS) notation</a> on legal characters for the consensus secondary structure line <em>SS_cons</em> and their interpretation</dd></dl>
<h2><a class="anchor" id="msa-formats-fasta"></a>
FASTA (Pearson) format</h2>
<dl class="section note"><dt>Note</dt><dd>Sequence names must not contain whitespace characters. Otherwise, the parts after the first whitespace will be dropped. The only allowed gap character is the hyphen <em></em>("-").</dd></dl>
<p>Here is an example alignment in FASTA format: </p><pre class="fragment">&gt;AL031296.1/85969-86120
CUGCCUCACAACGUUUGUGCCUCAGUUACCCGUAGAUGUAGUGAGGGUAACAAUACUUAC
UCUCGUUGGUGAUAAGGAACAGCU
&gt;AANU01225121.1/438-603
CUGCCUCACAACAUUUGUGCCUCAGUUACUCAUAGAUGUAGUGAGGGUGACAAUACUUAC
UCUCGUUGGUGAUAAGGAACAGCU
&gt;AAWR02037329.1/29294-29150
---CUCGACACCACU---GCCUCGGUUACCCAUCGGUGCAGUGCGGGUAGUAGUACCAAU
GCUAAUUAGUUGUGAGGACCAACU
</pre><h2><a class="anchor" id="msa-formats-maf"></a>
MAF format</h2>
<p>The multiple alignment format (MAF) is usually used to store multiple alignments on DNA level between entire genomes. It consists of independent blocks of aligned sequences which are annotated by their genomic location. Consequently, an MAF formatted MSA file may contain multiple records. MAF files start with a line </p><pre class="fragment">##maf
</pre><p> which is optionally extended by whitespace delimited key=value pairs. Lines starting with the character ("#") are considered comments and usually ignored.</p>
<p>A MAF block starts with character ("a") at the beginning of a line, optionally followed by whitespace delimited key=value pairs. The next lines start with character ("s") and contain sequence information of the form </p><pre class="fragment">s src start size strand srcSize sequence
</pre><p> where</p><ul>
<li><em>src</em> is the name of the sequence source</li>
<li><em>start</em> is the start of the aligned region within the source (0-based)</li>
<li><em>size</em> is the length of the aligned region without gap characters</li>
<li><em>strand</em> is either ("+") or ("-"), depicting the location of the aligned region relative to the source</li>
<li><em>srcSize</em> is the size of the entire sequence source, e.g. the full chromosome</li>
<li><em>sequence</em> is the aligned sequence including gaps depicted by the hyphen ("-")</li>
</ul>
<p>Here is an example alignment in MAF format (bluntly taken from the <a href="https://cgwb.nci.nih.gov/FAQ/FAQformat.html#format5">UCSC Genome browser website</a>): </p><pre class="fragment">##maf version=1 scoring=tba.v8 
# tba.v8 (((human chimp) baboon) (mouse rat)) 
# multiz.v7
# maf_project.v5 _tba_right.maf3 mouse _tba_C
# single_cov2.v4 single_cov2 /dev/stdin
                   
a score=23262.0     
s hg16.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG
                   
a score=5062.0                    
s hg16.chr7    27699739 6 + 158545518 TAAAGA
s panTro1.chr6 28862317 6 + 161576975 TAAAGA
s baboon         241163 6 +   4622798 TAAAGA 
s mm4.chr6     53303881 6 + 151104725 TAAAGA
s rn3.chr4     81444246 6 + 187371129 taagga

a score=6636.0
s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
s baboon         249182 13 +   4622798 gcagctgaaaaca
s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA

</pre><h1><a class="anchor" id="constraint-formats"></a>
File formats to manipulate the RNA folding grammar</h1>
<h2><a class="anchor" id="constraint-formats-file"></a>
Command Files</h2>
<p>The RNAlib and many programs of the ViennaRNA Package can parse and apply data from so-called command files. These commands may refer to structure constraints or even extensions of the RNA folding grammar (such as <a class="el" href="group__domains__up.html">Unstructured Domains</a>). Commands are given as a line of whitespace delimited data fields. The syntax we use extends the constraint definitions used in the <a href="http://mfold.rna.albany.edu/?q=mfold">mfold</a> / <a href="http://mfold.rna.albany.edu/?q=DINAMelt/software">UNAfold</a> software, where each line begins with a command character followed by a set of positions.<br  />
However, we introduce several new commands, and allow for an optional loop type context specifier in form of a sequence of characters, and an orientation flag that enables one to force a nucleotide to pair upstream, or downstream.</p>
<h3><a class="anchor" id="constraint_commands"></a>
Constraint commands</h3>
<p>The following set of commands is recognized:</p><ul>
<li><code>F</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Force</li>
<li><code>P</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Prohibit</li>
<li><code>C</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Conflicts/Context dependency</li>
<li><code>A</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Allow (for non-canonical pairs)</li>
<li><code>E</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Soft constraints for unpaired position(s), or base pair(s)</li>
</ul>
<h3><a class="anchor" id="domain_commands"></a>
RNA folding grammar exensions</h3>
<ul>
<li><code>UD</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Add ligand binding using the <a class="el" href="group__domains__up.html">Unstructured Domains</a> feature</li>
</ul>
<h3><a class="anchor" id="command_file_loop_types"></a>
Specification of the loop type context</h3>
<p>The optional loop type context specifier <code></code>[LOOP] may be a combination of the following:</p><ul>
<li><code>E</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Exterior loop</li>
<li><code>H</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Hairpin loop</li>
<li><code>I</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Interior loop</li>
<li><code>M</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Multibranch loop</li>
<li><code>A</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> All loops</li>
</ul>
<p>For structure constraints, we additionally allow one to address base pairs enclosed by a particular kind of loop, which results in the specifier <code></code>[WHERE] which consists of <code></code>[LOOP] plus the following character:</p><ul>
<li><code>i</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> enclosed pair of an Interior loop</li>
<li><code>m</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> enclosed pair of a Multibranch loop</li>
</ul>
<p>If no <code></code>[LOOP] or <code></code>[WHERE] flags are set, all contexts are considered (equivalent to <code>A</code> )</p>
<h3><a class="anchor" id="const_file_orientation"></a>
Controlling the orientation of base pairing</h3>
<p>For particular nucleotides that are forced to pair, the following <code></code>[ORIENTATION] flags may be used:</p><ul>
<li><code>U</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Upstream</li>
<li><code>D</code> <picture><source srcset="form_1_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ \ldots $" src="form_1.png" width="15" height="3"/></picture> Downstream</li>
</ul>
<p>If no <code></code>[ORIENTATION] flag is set, both directions are considered.</p>
<h3><a class="anchor" id="const_file_seq_coords"></a>
Sequence coordinates</h3>
<p>Sequence positions of nucleotides/base pairs are <picture><source srcset="form_2_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ 1- $" src="form_2.png" width="17" height="10"/></picture> based and consist of three positions <picture><source srcset="form_3_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ i $" src="form_3.png" width="6" height="10"/></picture>, <picture><source srcset="form_4_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ j $" src="form_4.png" width="7" height="12"/></picture>, and <picture><source srcset="form_5_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ k $" src="form_5.png" width="7" height="11"/></picture>. Alternativly, four positions may be provided as a pair of two position ranges <picture><source srcset="form_6_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ [i:j] $" src="form_6.png" width="29" height="15"/></picture>, and <picture><source srcset="form_7_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ [k:l] $" src="form_7.png" width="30" height="15"/></picture> using the '-' sign as delimiter within each range, i.e. <picture><source srcset="form_8_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ i-j $" src="form_8.png" width="29" height="12"/></picture>, and <picture><source srcset="form_9_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ k-l $" src="form_9.png" width="29" height="11"/></picture>.</p>
<h3><a class="anchor" id="const_file_syntax"></a>
Valid constraint commands</h3>
<p>Below are resulting general cases that are considered <em>valid</em> constraints:</p>
<ol type="1">
<li><b>"Forcing a range of nucleotide positions to be paired"</b>:<br  />
 Syntax: <div class="fragment"><div class="line">F i 0 k [WHERE] [ORIENTATION] </div>
</div><!-- fragment --><br  />
 Description:<br  />
 Enforces the set of <picture><source srcset="form_5_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ k $" src="form_5.png" width="7" height="11"/></picture> consecutive nucleotides starting at position <picture><source srcset="form_3_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ i $" src="form_3.png" width="6" height="10"/></picture> to be paired. The optional loop type specifier <code></code>[WHERE] allows to force them to appear as closing/enclosed pairs of certain types of loops.</li>
<li><b>"Forcing a set of consecutive base pairs to form"</b>:<br  />
 Syntax: <pre class="fragment">F i j k [WHERE] </pre><br  />
 Description:<br  />
 Enforces the base pairs <picture><source srcset="form_10_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j), \ldots, (i+(k-1), j-(k-1)) $" src="form_10.png" width="202" height="15"/></picture> to form. The optional loop type specifier <code></code>[WHERE] allows to specify in which loop context the base pair must appear.</li>
<li><b>"Prohibiting a range of nucleotide positions to be paired"</b>:<br  />
 Syntax: <pre class="fragment">P i 0 k [WHERE] </pre><br  />
 Description:<br  />
 Prohibit a set of <picture><source srcset="form_5_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ k $" src="form_5.png" width="7" height="11"/></picture> consecutive nucleotides to participate in base pairing, i.e. make these positions unpaired. The optional loop type specifier <code></code>[WHERE] allows to force the nucleotides to appear within the loop of specific types.</li>
<li><b>"Probibiting a set of consecutive base pairs to form"</b>:<br  />
 Syntax: <pre class="fragment">P i j k [WHERE] </pre><br  />
 Description:<br  />
 Probibit the base pairs <picture><source srcset="form_10_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j), \ldots, (i+(k-1), j-(k-1)) $" src="form_10.png" width="202" height="15"/></picture> to form. The optional loop type specifier <code></code>[WHERE] allows to specify the type of loop they are disallowed to be the closing or an enclosed pair of.</li>
<li><b>"Prohibiting two ranges of nucleotides to pair with each other"</b>:<br  />
 Syntax: <pre class="fragment">P i-j k-l [WHERE] </pre> Description:<br  />
 Prohibit any nucleotide <picture><source srcset="form_11_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ p \in [i:j] $" src="form_11.png" width="54" height="15"/></picture> to pair with any other nucleotide <picture><source srcset="form_12_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ q \in [k:l] $" src="form_12.png" width="54" height="15"/></picture>. The optional loop type specifier <code></code>[WHERE] allows to specify the type of loop they are disallowed to be the closing or an enclosed pair of.</li>
<li><b>"Enforce a loop context for a range of nucleotide positions"</b>:<br  />
 Syntax: <pre class="fragment">C i 0 k [WHERE] </pre> Description:<br  />
 This command enforces nucleotides to be unpaired similar to <em>prohibiting</em> nucleotides to be paired, as described above. It too marks the corresponding nucleotides to be unpaired, however, the <code></code>[WHERE] flag can be used to enforce specfic loop types the nucleotides must appear in.</li>
<li><b>"Remove pairs that conflict with a set of consecutive base pairs"</b>:<br  />
 Syntax: <pre class="fragment">C i j k </pre><br  />
 Description:<br  />
 Remove all base pairs that conflict with a set of consecutive base pairs <picture><source srcset="form_10_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j), \ldots, (i+(k-1), j-(k-1)) $" src="form_10.png" width="202" height="15"/></picture>. Two base pairs <picture><source srcset="form_13_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j) $" src="form_13.png" width="26" height="15"/></picture> and <picture><source srcset="form_14_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (p,q) $" src="form_14.png" width="29" height="15"/></picture> conflict with each other if <picture><source srcset="form_15_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ i &lt; p &lt; j &lt; q $" src="form_15.png" width="81" height="12"/></picture>, or <picture><source srcset="form_16_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ p &lt; i &lt; q &lt; j $" src="form_16.png" width="80" height="12"/></picture>.</li>
<li><b>"Allow a set of consecutive (non-canonical) base pairs to form"</b>:<br  />
 Syntax: <div class="fragment"><div class="line">A i j k [WHERE] </div>
</div><!-- fragment --><br  />
 Description:<br  />
 This command enables the formation of the consecutive base pairs <picture><source srcset="form_10_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j), \ldots, (i+(k-1), j-(k-1)) $" src="form_10.png" width="202" height="15"/></picture>, no matter if they are <em>canonical</em>, or <em>non-canonical</em>. In contrast to the above <code>F</code> and <code>W</code> commands, which remove conflicting base pairs, the <code>A</code> command does not. Therefore, it may be used to allow <em>non-canoncial</em> base pair interactions. Since the RNAlib does not contain free energy contributions <picture><source srcset="form_17_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ E_{ij} $" src="form_17.png" width="18" height="15"/></picture> for non-canonical base pairs <picture><source srcset="form_13_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j) $" src="form_13.png" width="26" height="15"/></picture>, they are scored as the <em>maximum</em> of similar, known contributions. In terms of a <em>Nussinov</em> like scoring function the free energy of non-canonical base pairs is therefore estimated as <p class="formulaDsp">
<picture><source srcset="form_18_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaDsp" alt="\[ E_{ij} = \min \left[  \max_{(i,k) \in \{GC, CG, AU, UA, GU, UG\}} E_{ik}, \max_{(k,j) \in \{GC, CG, AU, UA, GU, UG\}} E_{kj} \right]. \]" src="form_18.png" width="448" height="34"/></picture>
</p>
 The optional loop type specifier <code></code>[WHERE] allows to specify in which loop context the base pair may appear.</li>
<li><b>"Apply pseudo free energy to a range of unpaired nucleotide positions"</b>:<br  />
 Syntax: <div class="fragment"><div class="line">E i 0 k e </div>
</div><!-- fragment --><br  />
 Description:<br  />
 Use this command to apply a pseudo free energy of <picture><source srcset="form_19_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ e $" src="form_19.png" width="6" height="7"/></picture> to the set of <picture><source srcset="form_5_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ k $" src="form_5.png" width="7" height="11"/></picture> consecutive nucleotides, starting at position <picture><source srcset="form_3_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ i $" src="form_3.png" width="6" height="10"/></picture>. The pseudo free energy is applied only if these nucleotides are considered unpaired in the recursions, or evaluations, and is expected to be given in <picture><source srcset="form_20_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ kcal / mol $" src="form_20.png" width="55" height="15"/></picture>.</li>
<li><b>"Apply pseudo free energy to a set of consecutive base pairs"</b>:<br  />
 Syntax <div class="fragment"><div class="line">E i j k e </div>
</div><!-- fragment --><br  />
 Use this command to apply a pseudo free energy of <picture><source srcset="form_19_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ e $" src="form_19.png" width="6" height="7"/></picture> to the set of base pairs <picture><source srcset="form_10_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ (i,j), \ldots, (i+(k-1), j-(k-1)) $" src="form_10.png" width="202" height="15"/></picture>. Energies are expected to be given in <picture><source srcset="form_20_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ kcal / mol $" src="form_20.png" width="55" height="15"/></picture>.</li>
</ol>
<h3><a class="anchor" id="domains_syntax"></a>
Valid domain extensions commands</h3>
<ol type="1">
<li><b>"Add ligand binding to unpaired motif (a.k.a. unstructured domains)"</b>:<br  />
 Syntax: <div class="fragment"><div class="line">UD m e [LOOP] </div>
</div><!-- fragment --><br  />
 Description:<br  />
 Add ligand binding to unpaired sequence motif <picture><source srcset="form_21_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ m $" src="form_21.png" width="14" height="7"/></picture> (given in IUPAC format, capital letters) with binding energy <picture><source srcset="form_19_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ e $" src="form_19.png" width="6" height="7"/></picture> in particular loop type(s).<br  />
 Example: <div class="fragment"><div class="line">UD  AAA   -5.0    A</div>
</div><!-- fragment --><br  />
 The above example applies a binding free energy of <picture><source srcset="form_22_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ -5 kcal/mol $" src="form_22.png" width="73" height="15"/></picture> for a motif AAA that may be present in all loop types.</li>
</ol>
<h1><a class="anchor" id="energy-parameter-formats"></a>
File Formats for Energy Parameters</h1>
<h2><a class="anchor" id="modified-bases-params"></a>
JSON Parameter Files for Modified Bases</h2>
<p>The functions <a class="el" href="group__modified__bases.html#gac353e59d5f3176f5c1e59a93cc2b4d05" title="Prepare soft constraint callbacks for modified base as specified in JSON string.">vrna_sc_mod()</a>, <a class="el" href="group__modified__bases.html#ga48cef1af67413cc9bb4c82d6bc4010ad" title="Prepare soft constraint callbacks for modified base as specified in JSON string.">vrna_sc_mod_json()</a> and alike implement an energy correction framework to account for modified bases in the secondary structure predictions. To supply these functions with the energy parameters and general specifications of the base modification, the following JSON data format may be used:</p>
<p>JSON data must consist of a header section <b>modified_bases</b>. This header is an object with the mandatory keys:</p><ul>
<li><b>name</b> specifying a name of the modified base</li>
<li><b>unmodified</b> that consists of a single upper-case letter of the unmodified version of this base,</li>
<li>the <b>one_letter_code</b> key to specify which letter is used for the modified bases in the subsequent energy parameters, and</li>
<li>an array of <b>pairing_partners</b>.</li>
</ul>
<p>The latter must be uppercase characters. An optional <b>sources</b> key may contain an array of related publications, e.g. those the parameters have been derived from.</p>
<p>Next to the header may follow additional keys to specify the actual energy contributions of the modified base in various loop contexts. All energy contributions must be specified in free energies <picture><source srcset="form_23_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$\Delta G$" src="form_23.png" width="22" height="11"/></picture> in units of <picture><source srcset="form_24_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ kcal \cdot mol^{-1}$" src="form_24.png" width="72" height="12"/></picture>. To allow for rescaling of the free energies at temperatures that differ from the default ( <picture><source srcset="form_25_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$37^\circ C$" src="form_25.png" width="32" height="11"/></picture>), enthalpy parameters <picture><source srcset="form_26_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$\Delta H$" src="form_26.png" width="23" height="11"/></picture> may be specified as well. Those, however are optional. The keys for free energy (at <picture><source srcset="form_25_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$37^\circ C$" src="form_25.png" width="32" height="11"/></picture>) and enthalpy parameters have the suffixes <b>_energies</b> and <b>_enthalpies</b>, respectively.</p>
<p>The parser and underlying framework currently supports the following loop contexts:</p><ul>
<li>base pair stacks (via the <b>stacking</b> key prefix).<br  />
 This key must point to an object with one key value pair for each stacking interaction data is provided for. Here, the key consists of four upper-case characters denoting the interacting bases, where the the first two represent one strand in 5' to 3' direction and the last two the opposite strand in 3' to 5' direction. The values are energies in <picture><source srcset="form_24_dark.png" media="(prefers-color-scheme: dark)"/><img class="formulaInl" alt="$ kcal \cdot mol^{-1}$" src="form_24.png" width="72" height="12"/></picture>.</li>
<li>terminal mismatches (via the <b>mismatch</b> key prefix)<br  />
 This key points to an object with key value pairs for each mismatch energy parameter that is available. Keys are 4 characters long nucleotide one-letter codes as used in base pair stacks above. The second and fourth character denote the two unpaired mismatching bases, while the other two represent the closing base pair.</li>
<li>dangling ends (via the <b>dangle5</b> and <b>dangle3</b> key prefixes)<br  />
 The object behind these keys, again, consists of key value pairs for each dangling end energy parameter. Keys are 3 characters long where the first two represent the two nucleotides that form the base pair, and the third is the unpaired base that either stacks on the 3' or 5' end of the enclosed part of the base pair.</li>
<li>terminal pairs (via the <b>terminal</b> key prefix)<br  />
 Terminal base pairs, such as AU or GU, sometimes receive an additional energy penalty. The object behind this key may list energy parameters to apply whenever particular base pairs occur at the end of a helix. Each of those parameters is specified as key value pair, where the key consists of two upper-case characters denoting the terminal base pair.</li>
</ul>
<p>Below is a JSON template specifying most of the possible input parameters. Actual energy parameter files can be found in the source code tarball within the <b>misc/</b> subdirectory.</p>
<div class="fragment"><div class="line">{</div>
<div class="line">  &quot;modified_base&quot; : {</div>
<div class="line">    &quot;name&quot; : &quot;My modification (M)&quot;,</div>
<div class="line">    &quot;sources&quot; : [</div>
<div class="line">      {</div>
<div class="line">        &quot;authors&quot; : &quot;Author 1, Author 2&quot;,</div>
<div class="line">        &quot;title&quot; : &quot;UV-melting of modified oligos&quot;,</div>
<div class="line">        &quot;journal&quot; : &quot;Some journal&quot;,</div>
<div class="line">        &quot;year&quot; : 2022,</div>
<div class="line">        &quot;doi&quot; : &quot;10.0000/000000&quot;</div>
<div class="line">      }</div>
<div class="line">    ],</div>
<div class="line">    &quot;unmodified&quot; : &quot;G&quot;,</div>
<div class="line">    &quot;pairing_partners&quot; : [</div>
<div class="line">      &quot;U&quot;,&quot;A&quot;</div>
<div class="line">    ],</div>
<div class="line">    &quot;one_letter_code&quot; : &quot;M&quot;</div>
<div class="line">  },</div>
<div class="line">  &quot;stacking_energies&quot; : {</div>
<div class="line">    &quot;MAUU&quot; :  -1.2,</div>
<div class="line">    &quot;AGMC&quot; :  -2.73</div>
<div class="line">  },</div>
<div class="line">  &quot;stacking_enthalpies&quot; : {</div>
<div class="line">    &quot;MAUU&quot; :  -11.1,</div>
<div class="line">    &quot;AGMC&quot; :  -9.73</div>
<div class="line">  },</div>
<div class="line">  &quot;terminal_energies&quot; : {</div>
<div class="line">    &quot;MU&quot; : 0.5,</div>
<div class="line">    &quot;UM&quot; : 0.5</div>
<div class="line">  },</div>
<div class="line">  &quot;terminal_enthalpies&quot; : {</div>
<div class="line">    &quot;MU&quot; : 2.0,</div>
<div class="line">    &quot;UM&quot; : 2.0</div>
<div class="line">  },</div>
<div class="line">  &quot;mismatch_energies&quot; : {</div>
<div class="line">    &quot;CMGM&quot; : -1.11,</div>
<div class="line">    &quot;AGUM&quot; : -0.73</div>
<div class="line">  },</div>
<div class="line">  &quot;mismatch_enthalpies&quot; : {</div>
<div class="line">    &quot;CMGM&quot; : -11.11,</div>
<div class="line">    &quot;AGUM&quot; : -7.73</div>
<div class="line">  },</div>
<div class="line">  &quot;dangle5_energies&quot; : {</div>
<div class="line">    &quot;UAM&quot; : -1.01</div>
<div class="line">  },</div>
<div class="line">  &quot;dangle5_enthalpies&quot; : {</div>
<div class="line">    &quot;UAM&quot; : -6.01</div>
<div class="line">  },</div>
<div class="line">  &quot;dangle3_energies&quot; : {</div>
<div class="line">    &quot;CGM&quot; : -2.1,</div>
<div class="line">    &quot;GCM&quot; : -1.3</div>
<div class="line">  }</div>
<div class="line">}</div>
</div><!-- fragment --> <dl class="section see"><dt>See also</dt><dd><code>misc/rna_mod_template_parameters.json</code> in the source code tarball</dd></dl>
<p>An actual example of real-world data may look like </p><div class="fragment"><div class="line">{</div>
<div class="line">  &quot;modified_base&quot; : {</div>
<div class="line">    &quot;name&quot; : &quot;Pseudouridine&quot;,</div>
<div class="line">    &quot;sources&quot; : [</div>
<div class="line">      {</div>
<div class="line">        &quot;authors&quot;: &quot;Graham A. Hudson, Richard J. Bloomingdale, and Brent M. Znosko&quot;,</div>
<div class="line">        &quot;title&quot; : &quot;Thermodynamic contribution and nearest-neighbor parameters of pseudouridine-adenosine base pairs in oligoribonucleotides&quot;,</div>
<div class="line">        &quot;journal&quot; : &quot;RNA 19:1474-1482&quot;,</div>
<div class="line">        &quot;year&quot; : 2013,</div>
<div class="line">        &quot;doi&quot; : &quot;10.1261/rna.039610.113&quot;</div>
<div class="line">      }</div>
<div class="line">    ],</div>
<div class="line">    &quot;unmodified&quot; : &quot;U&quot;,</div>
<div class="line">    &quot;pairing_partners&quot; : [</div>
<div class="line">      &quot;A&quot;</div>
<div class="line">    ],</div>
<div class="line">    &quot;one_letter_code&quot; : &quot;P&quot;</div>
<div class="line">  },</div>
<div class="line">  &quot;stacking_energies&quot; : {</div>
<div class="line">    &quot;APUA&quot; :  -2.8,</div>
<div class="line">    &quot;CPGA&quot; : -2.77,</div>
<div class="line">    &quot;GPCA&quot; : -3.29,</div>
<div class="line">    &quot;UPAA&quot; : -1.62,</div>
<div class="line">    &quot;PAAU&quot; : -2.10,</div>
<div class="line">    &quot;PCAG&quot; : -2.49,</div>
<div class="line">    &quot;PGAC&quot; : -2.2,</div>
<div class="line">    &quot;PUAA&quot; : -2.74</div>
<div class="line">  },</div>
<div class="line">  &quot;stacking_enthalpies&quot; : {</div>
<div class="line">    &quot;APUA&quot; : -22.08,</div>
<div class="line">    &quot;CPGA&quot; : -16.23,</div>
<div class="line">    &quot;GPCA&quot; : -24.07,</div>
<div class="line">    &quot;UPAA&quot; : -20.81,</div>
<div class="line">    &quot;PAAU&quot; : -12.47,</div>
<div class="line">    &quot;PCAG&quot; : -17.29,</div>
<div class="line">    &quot;PGAC&quot; : -11.19,</div>
<div class="line">    &quot;PUAA&quot; : -26.94</div>
<div class="line">  },</div>
<div class="line">  &quot;terminal_energies&quot; : {</div>
<div class="line">    &quot;PA&quot; : 0.31,</div>
<div class="line">    &quot;AP&quot; : 0.31</div>
<div class="line">  },</div>
<div class="line">  &quot;terminal_enthalpies&quot; : {</div>
<div class="line">    &quot;PA&quot; : -2.04,</div>
<div class="line">    &quot;AP&quot; : -2.04</div>
<div class="line">  }</div>
<div class="line">}</div>
</div><!-- fragment --> <dl class="section see"><dt>See also</dt><dd><code>misc/rna_mod_pseudouridine_parameters.json</code> in the source code tarball </dd></dl>
</div></div><!-- contents -->
</div><!-- PageDoc -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Sun Apr 2 2023 13:14:55 for RNAlib-2.6.0b by <a href="https://www.doxygen.org/index.html"><img class="footer" src="doxygen.svg" width="104" height="31" alt="doxygen"/></a> 1.9.6 </li>
  </ul>
</div>
</body>
</html>
